tiledb/sm/filesystem/s3.cc

```diff
@@ -511,6 +513,7 @@ void S3::touch(const URI& uri) const {
   if (!s3_params_.sse_kms_key_id_.empty())
     put_object_request.SetSSEKMSKeyId(
         Aws::String(s3_params_.sse_kms_key_id_.c_str()));
+  put_object_request.SetStorageClass(storage_class_);
```

Comment:
We don't need an `if (storage_class_ != NOT_SET)` here, the SDK has this logic by itself.

https://github.com/aws/aws-sdk-cpp/blob/f067d450a8689f3ae05fbcd96039cdd9f2d0276c/generated/src/aws-cpp-sdk-s3/source/model/CreateMultipartUploadRequest.cpp#L181

Reply:
Turns out we do; that logic was added in a subsequent SDK version.